### PR TITLE
Implemented an example file for testing

### PR DESCRIPTION
**Pull request by alanpapalia**
- Added an example
-  made some runtime improvements so can quickly test example

### CODE REVIEW

**Comment by alanpapalia** [comment]
Ready for review @TimMagoun if you're mentally prepared to look at some performant code :sunglasses:

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,23 +9,28 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
...
 # Directory for built executables
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin CACHE PATH "Output directory for executables")
 
 # BUILD CONFIGURATIONS
 # Enable faster instruction sets (SIMD/AVX)
-set(ENABLE_VECTORIZATION OFF CACHE BOOL "Enable vectorized instruction sets (SIMD/AVX)? [disabled by default]")
+set(ENABLE_VECTORIZATION ON CACHE BOOL "Enable vectorized instruction sets (SIMD/AVX)? [enabled by default]")
 # Enable OpenMP (if available)
-set(ENABLE_OPENMP ON CACHE BOOL "Enable OpenMP (if available)")
+set(ENABLE_OPENMP OFF CACHE BOOL "Enable OpenMP (if available)")
```
Comment (TimMagoun) [inline CMakeLists.txt]: Consider removing openmp entirely?

File: src/CORA.cpp [hunk 2]
```diff
@@ -63,15 +83,37 @@ CoraTntResult solveCORA(Problem &problem, const Matrix &x0,
...
         f, QM, metric, retract, X, NablaF_Y, precon, params, user_function);
+    printIfVerbose(verbose, "Obtained solution with objective value: " +
+                                std::to_string(result.f));
 
     // check if the solution is certified
-    CertResults cert_results = problem.certify_solution(result.x, 1e-6, 10);
+    eta = result.f * rel_cert_eta;
+    eta = std::min(1e-1, eta);
+    eta = std::max(1e-6, eta);
+    cert_results = problem.certify_solution(result.x, eta, 10);
```
Comment (TimMagoun) [inline src/CORA.cpp]: What is the 10?

File: src/CORA.cpp [hunk 3]
```diff
@@ -63,15 +83,37 @@ CoraTntResult solveCORA(Problem &problem, const Matrix &x0,
...
     result = Optimization::Riemannian::TNT<Matrix, Matrix, Scalar, Matrix>(
         f, QM, metric, retract, X, NablaF_Y, precon, params, user_function);
+    printIfVerbose(verbose, "Obtained solution with objective value: " +
+                                std::to_string(result.f));
 
     // check if the solution is certified
-    CertResults cert_results = problem.certify_solution(result.x, 1e-6, 10);
+    eta = result.f * rel_cert_eta;
+    eta = std::min(1e-1, eta);
+    eta = std::max(1e-6, eta);
```
Comment (TimMagoun) [inline src/CORA.cpp]: 1e-1, 1e-6 could be consts in header?

File: src/CORA.cpp [hunk 4]
```diff
@@ -80,28 +122,38 @@ CoraTntResult solveCORA(Problem &problem, const Matrix &x0,
 
     // otherwise, increment the relaxation rank and try again
     problem.incrementRank();
-    Scalar grad_tol = 1e-6;
-    Scalar precon_grad_tol = 1e-6;
+    Scalar grad_tol = 1e-4;
+    Scalar precon_grad_tol = 1e-4;
```
Comment (TimMagoun) [inline src/CORA.cpp]: Could be consts

File: src/CORA.cpp [hunk 5]
```diff
@@ -80,28 +122,38 @@ CoraTntResult solveCORA(Problem &problem, const Matrix &x0,
...
     result = Optimization::Riemannian::TNT<Matrix, Matrix, Scalar, Matrix>(
         f, QM, metric, retract, X, NablaF_Y, precon, params, user_function);
+    printIfVerbose(verbose, "Obtained solution with objective value: " +
+                                std::to_string(result.f));
+
+    // let's check if the solution is certified
+    eta = result.f * rel_cert_eta;
+    eta = std::min(1e-1, eta);
+    eta = std::max(1e-6, eta);
+    cert_results = problem.certify_solution(result.x, eta, 10);
```
Comment (TimMagoun) [inline src/CORA.cpp]: Repeated code from lines 103?

File: src/CORA.cpp [hunk 6]
```diff
@@ -149,7 +201,7 @@ Matrix saddleEscape(const Problem &problem, const Matrix &Y, Scalar theta,
   // steplength,
   Scalar alpha_min = 1e-6; // Minimum stepsize
```
Comment (TimMagoun) [inline src/CORA.cpp:213]: Another const param for header file?

**Review by TimMagoun — COMMENTED** [review]
Looks good, a couple small comments about magic numbers and a repeated section of code

**Comment by alanpapalia** [comment]
reworked it to fix magic numbers, but since the vals are only used inside a single function, I preferred to keep them inside the source file instead of the header file.

Also, entirely removed OpenMP -- good call!

**Pull request merged**